yb-voyager/cmd/assessMigrationCommandV2.go

```diff
@@ -0,0 +1,424 @@
+		if err != nil {
+			return assessmentIssues, fmt.Errorf("failed to fetch unsupported queries on YugabyteDB: %w", err)
+		}
+		assessmentIssues = append(assessmentIssues, issues...)
+	}
+
+	unsupportedDataTypes, unsupportedDataTypesForLiveMigration, unsupportedDataTypesForLiveMigrationWithFForFB, err := fetchColumnsWithUnsupportedDataTypes()
+	if err != nil {
+		return assessmentIssues, fmt.Errorf("failed to fetch columns with unsupported data types: %w", err)
+	}
+
+	assessmentIssues = append(assessmentIssues, getAssessmentIssuesForUnsupportedDatatypes(unsupportedDataTypes)...)
+
+	assessmentIssues = append(assessmentIssues, fetchMigrationCaveatAssessmentIssues(unsupportedDataTypesForLiveMigration, unsupportedDataTypesForLiveMigrationWithFForFB)...)
+
+	return assessmentIssues, nil
+}
+
+// TODO: replacement of getAssessmentReportContentFromAnalyzeSchema()
+func getAssessmentReportContentFromAnalyzeSchemaV2() []AssessmentIssue {
+	/*
+		Here we are generating analyze schema report which converts issue instance to analyze schema issue
+		Then in assessment codepath we extract the required information from analyze schema issue which could have been done directly from issue instance(TODO)
+
+		But current Limitation is analyze schema currently uses regexp etc to detect some issues(not using parser).
+	*/
+	schemaAnalysisReport := analyzeSchemaInternal(&source, true)
+	assessmentReport.MigrationComplexity = schemaAnalysisReport.MigrationComplexity
+	assessmentReport.SchemaSummary = schemaAnalysisReport.SchemaSummary
+	assessmentReport.SchemaSummary.Description = lo.Ternary(source.DBType == ORACLE, SCHEMA_SUMMARY_DESCRIPTION_ORACLE, SCHEMA_SUMMARY_DESCRIPTION)
+
+	var assessmentIssues []AssessmentIssue
+	switch source.DBType {
+	case ORACLE:
+		assessmentIssues = append(assessmentIssues, fetchUnsupportedOracleFeaturesFromSchemaReportV2(schemaAnalysisReport)...)
+	case POSTGRESQL:
+		assessmentIssues = append(assessmentIssues, fetchUnsupportedPGFeaturesFromSchemaReportV2(schemaAnalysisReport)...)
+	default:
+		panic(fmt.Sprintf("unsupported source db type %q", source.DBType))
+	}
+
+	// Ques: Do we still need this and REPORT_UNSUPPORTED_QUERY_CONSTRUCTS env var
+	if utils.GetEnvAsBool("REPORT_UNSUPPORTED_PLPGSQL_OBJECTS", true) {
+		assessmentIssues = append(assessmentIssues, fetchUnsupportedPlPgSQLObjectsV2(schemaAnalysisReport)...)
+	}
+
+	return assessmentIssues
+}
+
+// TODO: will replace fetchUnsupportedOracleFeaturesFromSchemaReport()
+func fetchUnsupportedOracleFeaturesFromSchemaReportV2(schemaAnalysisReport utils.SchemaReport) []AssessmentIssue {
+	log.Infof("fetching assessment issues of feature category for Oracle...")
+	var assessmentIssues []AssessmentIssue
+	assessmentIssues = append(assessmentIssues, getAssessmentIssuesFromSchemaAnalysisReport(constants.FEATURE, "", COMPOUND_TRIGGER_ISSUE_REASON, schemaAnalysisReport, "")...)
+	return assessmentIssues
+}
+
+// TODO: will replace fetchUnsupportedPGFeaturesFromSchemaReport()
```

Comment:
> These functions will replace the existing codepath after UI updates are implemented.

@sanyamsinghal if we will be replacing the old functions after UI changes then how will it still work to get the previous structure as I believe we still need that for YugabyteD and maybe callhome. cc: @makalaaneesh

Reply:
hmm good catch, we might have to retain this then till the time ybd versions have moved to flattened AssessmentIssue approach.

To hide the current unflatten things in AssessmentReport struct, we can mark their json tag as `"-"` to avoid it getting in the assessmentReport.json

Reply:
we can't remove them from the json either, unfortunately 🥲 
because we send the json to yugabyted.

Reply:
yeah rights, thats bad, we send the complete assessment report json, not the require fields.